backend/de.metas.business.rest-api-impl/src/main/java/de/metas/rest_api/v1/bpartner/bpartnercomposite/jsonpersister/JsonPersisterService.java

```diff
@@ -1473,7 +1476,9 @@ private void syncJsonToLocation(
 			@NonNull final BPartnerLocation location,
 			@NonNull final SyncAdvise parentSyncAdvise)
 	{
-		final SyncAdvise syncAdvise = coalesce(jsonBPartnerLocation.getSyncAdvise(), parentSyncAdvise);
+		final BPartnerLocation originalBPartnerLocation = location.toBuilder().build();
```

Comment:
why are we doing toBuilder().build()?

Reply:
It's a way to clone the original object (without implementing an actual copy constructor), so as to later compare that no changes took place in this method (if using the `ASSERT_UNCHANGED` syncAdvice)

Reply:
ahh because the BPartnerLocation is not immutable!

ok, in this case would introduce BPartnerLocation.copy() method and do there toBuilder().build().

Why?
Because here you cannot decide if there are some other objects inside BPartnerLocation that need cloning too.
On a quick look, atm there are not, but you don;t know in future.
So, when they will be introduced, we just have to change the copy() method. 
If we won't have the copy() method, I am quite sure we will miss this hidden toBuilder().build() ....

Reply:
I missed deepCopy() entirely. Fixed.